packages/icon/package.py

```diff
@@ -1,294 +1,672 @@
-# Copyright 2013-2019 Lawrence Livermore National Security, LLC and other
-# Spack Project Developers. See the top-level COPYRIGHT file for details.
-#
-# SPDX-License-Identifier: (Apache-2.0 OR MIT)
-
 import os, subprocess
-from spack import *
+from collections import defaultdict
 
-import llnl.util.tty as tty
+from llnl.util import lang, filesystem
+from spack.util.environment import is_system_path, dump_environment
+from spack.util.executable import which_string
 
 
-class Icon(Package):
-    """The ICON modelling framework is a joint project between the
-    German Weather Service and the
-    Max Planck Institute for Meteorology for
-    developing a unified next-generation global numerical weather prediction and
-    climate modelling system. The ICON model has been introduced into DWD's
-    operational forecast system in January 2015."""
+class Icon(AutotoolsPackage):
+    """Icosahedral Nonhydrostatic Weather and Climate Model."""
 
-    homepage = 'https://gitlab.dkrz.de/icon/icon'
-    url = 'https://gitlab.dkrz.de/icon/icon/-/archive/icon-2.6.2.2/icon-icon-2.6.2.2.tar.gz'
+    homepage = 'https://code.mpimet.mpg.de/projects/iconpublic'
+    url = 'https://gitlab.dkrz.de/icon/icon/-/archive/icon-2.6.5.1/icon-icon-2.6.5.1.tar.gz'
     git = 'ssh://git@gitlab.dkrz.de/icon/icon.git'
 
-    maintainers = ['jonasjucker', 'PanicSheep']
-
-    version('master', branch='master', submodules=True)
-    version('2.6.5', tag='icon-2.6.5', submodules=True)
-    version('nwp',
-            git='ssh://git@gitlab.dkrz.de/icon/icon-nwp.git',
-            submodules=True)
-    version('cscs',
-            git='ssh://git@gitlab.dkrz.de/icon/icon-cscs.git',
-            submodules=True)
-    version('aes',
-            git='ssh://git@gitlab.dkrz.de/icon/icon-aes.git',
-            submodules=True)
-    version('ham',
-            git='ssh://git@git.iac.ethz.ch:hammoz/icon-hammoz.git',
-            branch='hammoz/gpu/master',
-            submodules=True)
+    version('develop', submodules=True)
+    version('2.6.5.1', tag='icon-2.6.5.1', submodules=True)
     version('exclaim-master',
             branch='master',
             git='ssh://git@github.com/C2SM/icon-exclaim.git',
             submodules=True)
-    version('c2sm-master',
-            branch='master',
-            git='ssh://git@github.com/C2SM/icon.git',
+    version('exclaim-test',
+            branch='test_spec',
+            git='ssh://git@github.com/C2SM/icon-exclaim.git',
+            submodules=True)
+    version('nwp-master',
+            git='ssh://git@gitlab.dkrz.de/icon/icon-nwp.git',
             submodules=True)
 
-    depends_on('cmake')
-    depends_on('libxml2')
-    depends_on('serialbox +fortran', when='serialize_mode=create')
-    depends_on('serialbox +fortran', when='serialize_mode=read')
-    depends_on('serialbox +fortran', when='serialize_mode=perturb')
-    depends_on('eccodes +aec +fortran', when='+eccodes')
-    depends_on('cosmo-eccodes-definitions',
```

Comment:
This cannot be deleted. Our users depend on it.

Reply:
I think this was just renamed to `serialization`.

Reply:
I meant to reference only `depends_on('cosmo-eccodes-definitions'`

Reply:
But now, that i think of this.
Maybe we should not make icon depend on cosmo-eccodes-definitions. Because icon doesn't **need** them. That dependency can be injected at runtime via `spack load cosmo-eccodes-definitions @...`

Reply:
That would be an elegant solution 👍🏻

Reply:
I just noticed that with this approach it's way too simple to create an incompatibility with having icon with one eccodes version and cosmo-eccodes-definitions with an other eccodes version. An according to Jean-Marie this is very dangerous.

Reply:
What about making icon only dependent on the definitions in case a variant is set i.e. `+cosmo-definitions`?
This would not affect most of our users and only add the additional dep into icon if needed.

Reply:
Can we just keep it the way it was: `depends_on('cosmo-eccodes-definitions',type=('build', 'run'), when='+eccodes')`
We know it works and it's the same as in cosmo and int2lm.